File: backend/app/Models/Facility.ts
import { DateTime } from 'luxon'
import { BaseModel, belongsTo, BelongsTo, column } from '@ioc:Adonis/Lucid/Orm'
import FacilityType from 'App/Models/FacilityType'

export default class Facility extends BaseModel {
	@column({ isPrimary: true })
	public id: number

	@column()
	public name: string

	@column()
	public address: string

	@column()
	public address2: string

	@column()
	public zip?: string

	@column()
	public city: string

	@column()
	public state: string

	@column()
	public countryId: string

	@column()
	public latitude?: number

	@column()
	public longitude?: number

	@column()
	public facilityTypeId: string

	@column()
	public phone?: string

	@column()
	public email?: string

	@column()
	public representativeName?: string

	@column()
	public representativePhone?: string

	@column()
	public representativeEmail?: string

	@column()
	public isActive: boolean

	@column.dateTime({ autoCreate: true })
	public createdAt: DateTime

	@column.dateTime({ autoCreate: true, autoUpdate: true })
	public updatedAt: DateTime

	@belongsTo(() => FacilityType, {
		localKey: 'id',
		foreignKey: 'facilityTypeId',
	})
	public type: BelongsTo<typeof FacilityType>
}

File: mobile/lib/data/models/entities/facility.dart
import '../base_entity.dart';

class Facility extends BaseEntity {
  @override
  int? id;
  String? name;
  String? address;
  String? address2;
  String? zip;
  String? city;
  String? state;
  int? countryId;
  String? latitude;
  String? longitude;
  int? facilityTypeId;
  String? phone;
  String? email;
  String? representativeName;
  String? representativePhone;
  String? representativeEmail;

  Facility({
    this.id,
    this.name,
    this.address,
    this.address2,
    this.zip,
    this.city,
    this.state,
    this.countryId,
    this.latitude,
    this.longitude,
    this.facilityTypeId,
    this.phone,
    this.email,
    this.representativeName,
    this.representativePhone,
    this.representativeEmail,
  }) : super(id: id);

  static List<Facility>? fromJsonAsList(dynamic data) {
    if (data == null) {
      return null;
    }
    return List<Facility>.from(data.map((x) => Facility.fromJson(x)));
  }

  Facility.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    address = json['address'];
    address2 = json['address2'];
    zip = json['zip'];
    city = json['city'];
    state = json['state'];
    countryId = json['country_id'];
    latitude = json['latitude'];
    longitude = json['longitude'];
    facilityTypeId = json['facility_type_id'];
    phone = json['phone'];
    email = json['email'];
    representativeName = json['representative_name'];
    representativePhone = json['representative_phone'];
    representativeEmail = json['representative_email'];
  }

  @override
  Map<String, dynamic> toJson() {
    final data = super.toJson();
    data['name'] = name;
    data['address'] = address;
    data['address2'] = address2;
    data['zip'] = zip;
    data['city'] = city;
    data['state'] = state;
    data['country_id'] = countryId;
    data['latitude'] = latitude;
    data['longitude'] = longitude;
    data['facility_type_id'] = facilityTypeId;
    data['phone'] = phone;
    data['email'] = email;
    data['representative_name'] = representativeName;
    data['representative_phone'] = representativePhone;
    data['representative_email'] = representativeEmail;
    return data;
  }
}
